src/colvarcomp_gpath.cpp

```diff
@@ -0,0 +1,826 @@
+            // Compute factors for polynomial combinations
+            cvm::real factor_polynomial = getPolynomialFactorOfCVGradient(i_cv);
+            for (size_t j_elem = 0; j_elem < cv[i_cv]->value().size(); ++j_elem) {
+                // Apply the gradients to the atom groups in i-th cv
+                // Loop over all atom groups
+                for (size_t k_ag = 0 ; k_ag < cv[i_cv]->atom_groups.size(); ++k_ag) {
+                    // Loop over all atoms in the k-th atom group
+                    for (size_t l_atom = 0; l_atom < (cv[i_cv]->atom_groups)[k_ag]->size(); ++l_atom) {
+                        // Chain rule
+                        (*(cv[i_cv]->atom_groups)[k_ag])[l_atom].grad = factor_polynomial * ((*(cv[i_cv]->atom_groups)[k_ag])[l_atom].grad * tmp_cv_grad_v1[j_elem] + (*(cv[i_cv]->atom_groups)[k_ag])[l_atom].grad * tmp_cv_grad_v2[j_elem]);
+                    }
+                }
+            }
+        }
+    }
+}
+
+void colvar::gzpathCV::apply_force(colvarvalue const &force) {
+    for (size_t i_cv = 0; i_cv < cv.size(); ++i_cv) {
+        // If this CV us explicit gradients, then atomic gradients is already calculated
+        // We can apply the force to atom groups directly
+        if ( cv[i_cv]->is_enabled(f_cvc_explicit_gradient) &&
+            !cv[i_cv]->is_enabled(f_cvc_scalable) &&
+            !cv[i_cv]->is_enabled(f_cvc_scalable_com)) {
+            for (size_t k_ag = 0 ; k_ag < cv[i_cv]->atom_groups.size(); ++k_ag) {
+                (cv[i_cv]->atom_groups)[k_ag]->apply_colvar_force(force.real_value);
+            }
+        }
+        else {
+            colvarvalue tmp_cv_grad_v1 = -1.0 * dzdv1[i_cv];
+            colvarvalue tmp_cv_grad_v2 =  1.0 * dzdv2[i_cv];
+            // Temporary variables storing gradients
+            // Compute factors for polynomial combinations
+            cvm::real factor_polynomial = getPolynomialFactorOfCVGradient(i_cv);
+            colvarvalue cv_force = force.real_value * factor_polynomial * (tmp_cv_grad_v1 + tmp_cv_grad_v2);
+            cv[i_cv]->apply_force(cv_force);
+        }
+    }
+}
+
+void GeometricPathCV::split_string(const std::string& data, const std::string& delim, std::vector<std::string>& dest) {
+    size_t index = 0, new_index = 0;
+    std::string tmpstr;
+    while (index != data.length()) {
+        new_index = data.find(delim, index);
+        if (new_index != std::string::npos) tmpstr = data.substr(index, new_index - index);
+        else tmpstr = data.substr(index, data.length());
+        if (!tmpstr.empty()) {
+            dest.push_back(tmpstr);
+        }
+        if (new_index == std::string::npos) break;
+        index = new_index + 1;
+    }
+}
+
+void GeometricPathCV::init_string_cv_map(std::map<std::string, std::function<colvar::cvc* (const std::string& subcv_conf)>>& string_cv_map) {
+    string_cv_map["distance"]              = [](const std::string& conf){return new colvar::distance(conf);};
```

Comment:
Not sure we can use `std::function`, and as a result, the map of lambda functions.

That said, I think this map (or a variant with old-style function pointers) should exist somewhere in the core of Colvars, it would be very useful to allow for a generic treatment of CVCs. This could be created by a variant of `colvar::init_components_type`. Right now the various instances of that function are called by `colvar::init_components`, but something similar could happen when initializing the module as a whole.

Reply:
`std::function` is new in C++11 and the code there should be replaced by functors. I think there should be a global `map` object linking `function_type` of each CVC to their clone function. `init_components_type` and `init_components` may have done similar things, but I am not sure about this.

Reply:
I would definitely welcome using this map in the core code, leaving it pretty much the same (`std::function` is much more readable than function pointers). There is no immediate need to change the `init_components_type` template, so the map could be used for now only in C++11-dependent features, such as this one. If things clear up a bit on the VMD side in the future, we may be able to switch to requiring C++11 for everything.

Reply:
My hope in the short term would be to lift the C++11 requirement for this as well, though. I didn't have in mind to further integrate C++11-only features... That would create a two-tier situation depending on platforms and builds, that could last for an unknown number of years.

Reply:
I agree to keep trying to support C++03 in the short term, but it has to be for reasonably small effort. How much that is is up to @HanatoK, at least until the PR is merged. After that, any small portability changes would be useful, and should be evaluated on a case-by-case basis.

Regarding when to start integrating C++11-only features: you have to start at some point.

Reply:
> I agree to keep trying to support C++03 in the short term, but it has to be for reasonably small effort. How much that is is up to @HanatoK, at least until the PR is merged. After that, any small portability changes would be useful, and should be evaluated on a case-by-case basis.

I expect significant portability fixes to happen mostly before the merge: the merge itself is a powerful incentive, that's what I'm trying to leverage here.

> Regarding when to start integrating C++11-only features: you have to start at some point.
Agreed, but it would be great if this could be somewhat synchronized with the major back-ends. I fear the increasing feature fragmentation. But yes, some of it is unavoidable. We have some now that we're trying to reduce (Lepton).

Reply:
@giacomofiorin @jhenin As for this map, I have tried to move it into `colvarmodule` but the issue is that it requires `colvar::cvc`, and `colvar::cvc` is a nested class in class `colvar`. The nested class `colvar::cvc` cannot be forward declared in `colvarmodule.h` due to C++ limitations (see https://stackoverflow.com/questions/1021793/how-do-i-forward-declare-an-inner-class). Do you have other ideas?

Reply:
@HanatoK Good point. I think that's the best way to go though, the `colvarmodule` object should have a map of components. That would make a lot of code more elegant and maintainable. I propose we un-nest the `cvc` class from `colvar` to allow a forward declaration. Thoughts, @giacomofiorin ?